### DISCUSSION TITLE
jxr formate processing problem on macos

### DISCUSSION THREAD

**Discussion by Zimovets** [ImageMagick/ImageMagick#7283] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

macOS 13.6.3 (22G436) Ventura

### Description

I am using imagemagick in my nodejs project i convert .jxr to jpeg, on my ubuntu i install it just apt install imagemagick, and everything works fine, but when my colleague run my code on the macOS we got an error, we try both approaches from docs how to install imagemagick, but it didn't help, how to fix it? problem only with jxt formate another formates i work with work properly. Here is an error:

   Error: Command failed: sh: JxrDecApp: command not found
mv: /var/folders/gx/_rbq2psn571_7kls1gkwrd7h0000gp/T/magick-fUOnqVeXBLB4sly4-LZyFOsHyUG9gFfA.tiff: No such file or directory
convert: delegate failed `/bin/mv '%i' '%i.jxr'; 'JxrDecApp' -i '%i.jxr' -o '%o.tiff'; /bin/mv '%i.jxr' '%i'; /bin/mv '%o.tiff' '%o'' @ error/delegate.c/InvokeDelegate/1924.
convert: unable to open file '/var/folders/gx/_rbq2psn571_7kls1gkwrd7h0000gp/T/magick-IS0fWoI1quFwR1eIs-Z5So3sstahisZh': No such file or directory @ error/constitute.c/ReadImage/786.
convert: no images defined `/Users/yuliiabila/projects/hillel-api/hillel-api/uploads/temp/temp_1714636178264.jpg' @ error/convert.c/ConvertImageCommand/3362.

    at ChildProcess.<anonymous> (/Users/yuliiabila/projects/hillel-api/hillel-api/node_modules/imagemagick/imagemagick.js:88:15)
    at ChildProcess.emit (node:events:514:28)
    at ChildProcess.emit (node:domain:489:12)
    at maybeClose (node:internal/child_process:1091:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
    at Process.callbackTrampoline (node:internal/async_hooks:130:17) {
  timedOut: false,
  killed: false,
  code: 1,
  signal: null
}
ERROR: Error: ENOENT: no such file or directory, unlink '/Users/yuliiabila/projects/hillel-api/hillel-api/uploads/temp/temp_1714636178264.jpg'

**Comment by snibgo** [comment]
>  /bin/mv '%i' '%i.jxr'; 'JxrDecApp' -i '%i.jxr' -o '%o.tiff'; /bin/mv '%i.jxr' '%i'; /bin/mv '%o.tiff' '%o''

One (or more) of those commands failed. I suppose `/bin/mv` is okay, so I would suspect `JxrDecApp`. Can that program be run on the macOS? If not, then that is the problem.

**Comment by dlemstra** [comment, marked as answer]
I think you need to install `jxrlib` to make sure `JxrDecApp` is available on the system.

**Reply by Zimovets** [reply]
it helped thanks a lot!

**Reply by Zimovets** [reply]
here is another question - https://github.com/ImageMagick/ImageMagick/discussions/7286 , maybe you can help
